src/python/pants/task/console_task.py

```diff
@@ -30,6 +30,12 @@ def register_options(cls, register):
 
   def __init__(self, *args, **kwargs):
     super(ConsoleTask, self).__init__(*args, **kwargs)
+    self._defines_console_output = not hasattr(self.console_output, '_canary')
```

Comment:
Is there a reason this checking isn't unified with the same thing in task.py's defines_ class methods?

Reply:
I... wasn't sure yet that this pattern was worth making generic. I don't know if I've seen it before, and I wasn't sure how folks would react to it.